Release notes for 8.16.0 (#16605)
elastic/logstash: docs/static/releasenotes.asciidoc

* <<logstash-8-0-0-alpha1,Logstash 8.0.0-alpha1>>

[[logstash-8-16-0]]
=== Logstash 8.16.0 Release Notes

==== Announcing the new {ls} Health Report API

> The new Health Report API (`GET /_health_report`) is available starting with {ls} `8.16.0`.
> This API leverages indicators capable of detecting the degraded status of pipelines and
> providing actionable insights https://github.com/elastic/logstash/pull/16520[#16520], https://github.com/elastic/logstash/pull/16532[#16532].

Reviewer: ```suggestion
The new Health Report API (`GET /_health_report`) is available starting with {ls} `8.16.0`.
This API uses indicators capable of detecting the degraded status of pipelines and
providing actionable insights https://github.com/elastic/logstash/pull/16520[#16520], https://github.com/elastic/logstash/pull/16532[#16532].
```

For a more detailed view please check <<logstash-health-report-api>>.

[[featured-8.16.0]]
==== New features and enhancements

> * {ls} now gracefully handles `if` conditionals that can't be evaluated (https://github.com/elastic/logstash/pull/16322[#16322]), either by dropping

Reviewer 2: ```suggestion
* {ls} now gracefully handles `if` conditionals in pipeline definitions that can't be evaluated (https://github.com/elastic/logstash/pull/16322[#16322]), either by dropping
```

> NOTE for users upgrading to {ls} 8.16.0: if your existing automation relies on liveliness scripts that expect Logstash's API status to either not be available or return a hardcoded `green` status, there is an escape hatch to prevent the new Health API's status from affecting the top-level `status` field of Logstash's existing API responses. Setting the `logstash.forceApiStatus` system property to `green` will force other APIs to return the previous hard-coded `green` value. https://github.com/elastic/logstash/pull/16535[#16535]

Reviewer: ```suggestion
If your {ls} instances use automation that relies on pre-8.16.0 behavior of API statuses, you can maintain the current behavior and prevent the new Health API status from affecting the top-level `status` field of existing {ls} API responses. 
You can add the `logstash.forceApiStatus: green` system property to the `logstash-yml` config file.
This setting forces other APIs to return the previous hard-coded `green` value. https://github.com/elastic/logstash/pull/16535[#16535]
```

Reviewer: Is it necessary to call out specific situations where this change might affect automations, or can we generalize as suggested?  I'm wondering how many users this will affect, and also thinking that uses affected would know based on the reduced description. Let's discuss. 

Also, we need to be specific about the location where the property should be added. I guessed. 
Let's discuss.

Reviewer 3: > Is it necessary to call out specific situations where this change might affect automations, or can we generalize as suggested?

IMO, I'd sticky to the more specific version. This change can cause troubles for users relying on that green status, specially the one using it for liveliness purposes. 

> Also, we need to be specific about the location where the property should be added.

Agreed, this seems to be a system property so I guess the correct place would be the `jvm.options` file, using the following format: `-Dlogstash.forceApiStatus=green`

@yaauie, @jsvd?

Reviewer 4: > Agreed, this seems to be a system property so I guess the correct place would be the jvm.options file, using the following format: -Dlogstash.forceApiStatus=green

Correct, from https://github.com/elastic/logstash/pull/16535:

> ### How to test this PR locally
>
> Add the following to your `config/jvm.options`:
> `-Dlogstash.forceApiStatus=green`

Reviewer 4: > IMO, I'd sticky to the more specific version. This change can cause troubles for users relying on that green status, specially the one using it for liveliness purposes.

+++ I'd like to keep a specific example so users understand the real life impact, as it may not be obvious at a first glance.

Reviewer: I updated the suggestion, taking discussion points into consideration. 
Please LMKWYT.

> For a more detailed view please check <<logstash-health-report-api>>.

Reviewer: ```suggestion
Check out the <<logstash-health-report-api>> docs more for info.
```

> [[featured-8.16.0]]

Reviewer: ```suggestion
[[featured-8-16-0]]
```

> The new Health Report API (`GET /_health_report`) is available starting with {ls} `8.16.0`.
> This API leverages indicators capable of detecting the degraded status of pipelines and
> providing actionable insights https://github.com/elastic/logstash/pull/16520[#16520], https://github.com/elastic/logstash/pull/16532[#16532].
> 
> For a more detailed view please check <<logstash-health-report-api>>.
> 
> [[featured-8.16.0]]
> ==== New features and enhancements
> 
> * {ls} now gracefully handles `if` conditionals in pipeline definitions that can't be evaluated (https://github.com/elastic/logstash/pull/16322[#16322]), either by dropping
> the event or by sending it to the pipeline's DLQ if enabled. https://github.com/elastic/logstash/pull/16423[#16423]
> 
> * Added an escape hatch to prevent the new Health API's status from affecting the top-level `status` field of Logstash's existing API responses. By setting the `logstash.forceApiStatus` system property to `green` you can force the other APIs to return the previous hard-coded `green` value. https://github.com/elastic/logstash/pull/16535[#16535]
> 

Reviewer 4: A final suggestion is to move the force api status setting closer to the feature announcement to increase changes of users spotting it while reading the release notes. I rephrased it a little bit and added more context on why you'd need to tweak it:

```suggestion
The new Health Report API (`GET /_health_report`) is available starting with {ls} `8.16.0`.
This API leverages indicators capable of detecting the degraded status of pipelines and
providing actionable insights https://github.com/elastic/logstash/pull/16520[#16520], https://github.com/elastic/logstash/pull/16532[#16532].

NOTE for users upgrading to {ls} 8.16.0: if your existing automation relies on liveliness scripts that expect Logstash's API status to either not be available or return a hardcoded `green` status, there is an escape hatch to prevent the new Health API's status from affecting the top-level `status` field of Logstash's existing API responses. Setting the `logstash.forceApiStatus` system property to `green` will force other APIs to return the previous hard-coded `green` value. https://github.com/elastic/logstash/pull/16535[#16535]

For a more detailed view please check <<logstash-health-report-api>>.

[[featured-8.16.0]]
==== New features and enhancements

* {ls} now gracefully handles `if` conditionals in pipeline definitions that can't be evaluated (https://github.com/elastic/logstash/pull/16322[#16322]), either by dropping
the event or by sending it to the pipeline's DLQ if enabled. https://github.com/elastic/logstash/pull/16423[#16423]

```

[[core-8.16.0]]
==== Other changes to Logstash core

* Added deprecation logs for modules `netflow`, `fb_apache` and `azure`. https://github.com/elastic/logstash/pull/16548[#16548]

* Added deprecation logs for users that doesn't explicitly select a value for `pipeline.buffer.type` forcing them to proactively make a choice before version `9.0` when this setting will default to heap. https://github.com/elastic/logstash/pull/16498[#16498]

* The flag `--event_api.tags.illegal` was deprecated and will be removed in version 9. This flag remains available throughout all version 8.x releases. Users who rely on this flag to allow non strings assignment to `tags` field should update their pipeline. https://github.com/elastic/logstash/pull/16507[#16507]

> ==== Plugins

Reviewer: ```suggestion
[[plugins-8-16-0]]
==== Plugins
```

*Cef Codec - 6.2.8*

> * Added missing documentation of the `raw_data_field` option https://github.com/logstash-plugins/logstash-codec-cef/pull/105[#105]

Reviewer: ```suggestion
* [DOC] Added missing documentation of the `raw_data_field` option https://github.com/logstash-plugins/logstash-codec-cef/pull/105[#105]
```

* <<logstash-8-0-0-alpha1,Logstash 8.0.0-alpha1>>

[[logstash-8-16-0]]
=== Logstash 8.16.0 Release Notes

> ==== Announcing the new {ls} health report API

Reviewer 4: ```suggestion
==== Announcing the new {ls} Health Report API
```

> The new health report API (`GET /_health_report`) is available with {ls} `8.16.0` and later,
> with indicators that are capable of detecting degraded status of configured pipelines and
> providing actionable insights https://github.com/elastic/logstash/pull/16520[#16520], https://github.com/elastic/logstash/pull/16532[#16532].

Reviewer 4: ```suggestion
The new Health Report API (`GET /_health_report`) is available starting with {ls} `8.16.0`.
This API leverages indicators capable of detecting the degraded status of pipelines and
providing actionable insights https://github.com/elastic/logstash/pull/16520[#16520], https://github.com/elastic/logstash/pull/16532[#16532].
```

For a more detailed view please check <<logstash-health-report-api>>.

[[notable-8.16.0]]
==== Notable issues fixed

> * When a conditional evaluation encounter an error in the expression, the event that triggered the issue is sent to pipeline's DLQ, if enabled for the executing pipeline. https://github.com/elastic/logstash/pull/16423[#16423]

Reviewer 3: It was merged with another related change and the section was renamed to "New features and enhancements": https://github.com/elastic/logstash/pull/16605/files/e7eac3fcacd5c4471f95e14c863e90f20d7cb88e#r1819116437

> * Fixed the `if` statement expression evaluation against runtime errors that could crash the pipeline, just logging the offending event and continuing with the next in the batch. https://github.com/elastic/logstash/pull/16322[#16322]

Reviewer 4: You can merge these two into a single entry, talking about how Logstash will now handle conditionals that can't be evaluated, either by dropping the event or sending to the DLQ if it's enabled <link1> <link2>

Reviewer 3: Done, could you please have a second look? Thanks!

[[core-8.16.0]]
==== Other changes to Logstash core

> * Added an escape hatch to prevent the new health API's status from affecting the status field of other API responses; by setting the `logstash.forceApiStatus` system property to `green`, you can force the other APIs to continue to use hard-coded green value. https://github.com/elastic/logstash/pull/16535[#16535]

Reviewer 4: ```suggestion
* Added an escape hatch to prevent the new Health API's status from affecting the top-level `status` field of Logstash's existing API responses. By setting the `logstash.forceApiStatus` system property to `green` you can force the other APIs to return the previous hard-coded `green` value. https://github.com/elastic/logstash/pull/16535[#16535]
```

* Added deprecation logs for modules `netflow`, `fb_apache` and `azure`. https://github.com/elastic/logstash/pull/16548[#16548]

> * Added deprecation logs for users that doesn't explicitly select a value for `pipeline.buffer.type` and resort to the default setting. https://github.com/elastic/logstash/pull/16498[#16498]

Reviewer 2: ```suggestion
* Added deprecation logs for users that doesn't explicitly select a value for `pipeline.buffer.type` forcing them to proactively make a a choice before version 9.0 when this setting will default to heap. https://github.com/elastic/logstash/pull/16498[#16498]
```

* The flag `--event_api.tags.illegal` was deprecated and will be removed in version 9. This flag remains available throughout all version 8.x releases. Users who rely on this flag to allow non strings assignment to `tags` field should update their pipeline. https://github.com/elastic/logstash/pull/16507[#16507]

==== Plugins

*Cef Codec - 6.2.8*

* Added missing documentation of the `raw_data_field` option https://github.com/logstash-plugins/logstash-codec-cef/pull/105[#105]

*Json_lines Codec - 3.2.2*

* Raised the default value of the `decode_size_limit_bytes` option to 512 MB https://github.com/logstash-plugins/logstash-codec-json_lines/pull/46[#46]

* Added the `decode_size_limit_bytes` option to limit the maximum size of JSON lines that can be parsed. https://github.com/logstash-plugins/logstash-codec-json_lines/pull/43[#43]

*Elastic_integration Filter - 0.1.15*

> * Applied Elasticsearch GeoIP changes and synced with Elasticsearch `8.16` https://github.com/elastic/logstash-filter-elastic_integration/pull/170[#170]

Reviewer 4: ```suggestion
* Use Elasticsearch code from its `8.16` branch and adapt to changes in Elasticsearch GeoIP processor https://github.com/elastic/logstash-filter-elastic_integration/pull/170[#170]
```

*Geoip Filter - 7.3.1*

> * Fixed to not crash pipelines when looking up a database with customised fields https://github.com/logstash-plugins/logstash-filter-geoip/pull/225[#225]

Reviewer 4: ```suggestion
* Fixed a pipeline crash when looking up a database with customised fields https://github.com/logstash-plugins/logstash-filter-geoip/pull/225[#225]
```

*Azure_event_hubs Input - 1.5.0*

* Updated Azure Event Hub client library to version `3.3.0` https://github.com/logstash-plugins/logstash-input-azure_event_hubs/pull/96[#96]

*Beats Input - 6.9.0*

> * Improved plugin's shutdown process and fixed to not crash when the connection is terminated while processing messages https://github.com/logstash-plugins/logstash-input-beats/pull/500[#500]

Reviewer 4: ```suggestion
* Improved plugin's shutdown process and fixed a crash when a connection is terminated while processing messages https://github.com/logstash-plugins/logstash-input-beats/pull/500[#500]
```

*Http Input - 3.9.1*

> * Fixed to correctly log the state of `ssl_enabled` during `run` https://github.com/logstash-plugins/logstash-input-http/pull/180[#180]

Reviewer 4: ```suggestion
* Fixed an issue where the value of `ssl_enabled` during `run` wasn't correctly logged https://github.com/logstash-plugins/logstash-input-http/pull/180[#180]
```
